pkg/chartrenderer/default.go

```diff
@@ -160,31 +160,48 @@ func (c *RenderedChart) Manifest() []byte {
 }
 
 // Files returns all rendered manifests mapping their names to their content.
-func (c *RenderedChart) Files() map[string]string {
-	var files = make(map[string]string)
+func (c *RenderedChart) Files() map[string]map[string]string {
+	var files = make(map[string]map[string]string)
 	for _, manifest := range c.Manifests {
-		files[manifest.Name] = manifest.Content
+		if _, ok := files[manifest.Name]; ok {
+			files[manifest.Name][strings.ToLower(manifest.Head.Kind+"/"+manifest.Head.Metadata.Name)] = manifest.Content
+			continue
+		}
+		files[manifest.Name] = map[string]string{strings.ToLower(manifest.Head.Kind + "/" + manifest.Head.Metadata.Name): manifest.Content}
 	}
 	return files
 }
 
 // FileContent returns explicitly the content of the provided <filename>.
 func (c *RenderedChart) FileContent(filename string) string {
+	var fileContent string
```

Comment:
You can use `strings.Builder` to speed up - string is immutable and every manipulation of it creates a new string.